nnstreamer_example/example_object_detection/nnstreamer_example_object_detection.cc

```diff
@@ -0,0 +1,738 @@
+
+  get_detected_objects (detections, boxes);
+
+  gst_memory_unmap (mem_boxes, &info_boxes);
+  gst_memory_unmap (mem_detections, &info_detections);
+
+  gst_memory_unref (mem_boxes);
+  gst_memory_unref (mem_detections);
+}
+
+/**
+ * @brief Set window title.
+ * @param name GstXImageSink element name
+ * @param title window title
+ */
+static void
+set_window_title (const gchar * name, const gchar * title)
+{
+  GstTagList *tags;
+  GstPad *sink_pad;
+  GstElement *element;
+
+  element = gst_bin_get_by_name (GST_BIN (g_app.pipeline), name);
+
+  g_return_if_fail (element != NULL);
+
+  sink_pad = gst_element_get_static_pad (element, "sink");
+
+  if (sink_pad) {
+    tags = gst_tag_list_new (GST_TAG_TITLE, title, NULL);
+    gst_pad_send_event (sink_pad, gst_event_new_tag (tags));
+    gst_object_unref (sink_pad);
+  }
+
+  gst_object_unref (element);
+}
+
+/**
+ * @brief Store the information from the caps that we are interested in.
+ */
+static void
+prepare_overlay_cb (GstElement * overlay, GstCaps * caps, gpointer user_data)
+{
+  CairoOverlayState *state = &g_app.overlay_state;
+
+  state->valid = gst_video_info_from_caps (&state->vinfo, caps);
+}
+
+/**
+ * @brief Callback to draw the overlay.
+ */
+static void
+draw_overlay_cb (GstElement * overlay, cairo_t * cr, guint64 timestamp,
+    guint64 duration, gpointer user_data)
+{
+  CairoOverlayState *state = &g_app.overlay_state;
+  gfloat x, y, width, height;
+  gchar *label;
+  guint drawed = 0;
+
+  g_return_if_fail (state->valid);
+  g_return_if_fail (g_app.running);
+
+  std::vector < DetectedObject > detected;
+  std::vector < DetectedObject >::iterator iter;
+
+  g_mutex_lock (&g_app.mutex);
+  detected = g_app.detected_objects;
+  g_mutex_unlock (&g_app.mutex);
+
+  /* set font props */
+  cairo_select_font_face (cr, "Sans", CAIRO_FONT_SLANT_NORMAL,
+      CAIRO_FONT_WEIGHT_BOLD);
+  cairo_set_font_size (cr, 20.0);
+
+  for (iter = detected.begin (); iter != detected.end (); ++iter) {
+    label =
+        (gchar *) g_list_nth_data (g_app.tflite_info.labels, iter->class_id);
+
+    x = iter->x * VIDEO_WIDTH / MODEL_WIDTH;
+    y = iter->y * VIDEO_HEIGHT / MODEL_HEIGHT;
+    width = iter->width * VIDEO_WIDTH / MODEL_WIDTH;
+    height = iter->height * VIDEO_WIDTH / MODEL_HEIGHT;
+
+    /* draw rectangle */
+    cairo_rectangle (cr, x, y, width, height);
+    cairo_set_source_rgb (cr, 1, 0, 0);
+    cairo_set_line_width (cr, 1.5);
+    cairo_stroke (cr);
+    cairo_fill_preserve (cr);
+
+    /* draw title */
+    cairo_move_to (cr, x + 5, y + 25);
+    cairo_text_path (cr, label);
+    cairo_set_source_rgb (cr, 1, 0, 0);
+    cairo_fill_preserve (cr);
+    cairo_set_source_rgb (cr, 1, 1, 1);
+    cairo_set_line_width (cr, .3);
+    cairo_stroke (cr);
+    cairo_fill_preserve (cr);
+
+    if (++drawed >= MAX_OBJECT_DETECTION) {
+      /* max objects drawed */
+      break;
+    }
+  }
+}
+
+/**
+ * @brief Main function.
+ */
+int
+main (int argc, char **argv)
```

Comment:
Which is the `standard` expression about pointer?
- `char **argv` or `char ** argv`

Reply:
I raised #686 for discussion about this problem.